2019-spring/01. Model fitting (NLLS)/nlls.tex

```diff
@@ -0,0 +1,793 @@
+
+  Тогда положим
+  \begin{equation*}
+    \begin{gathered}
+      \tilde{\mathbf{r}}=\frac{\sqrt{\rho'}}{1-\alpha}\mathbf{r} \\
+      \tilde{\mathbf{J}}=\sqrt{\rho'}\left(\mathbf{I}-\alpha\frac{\mathbf{r}\mathbf{r}^\intercal}{\left\Vert\mathbf{r}\right\Vert^2}\right)\mathbf{J}
+    \end{gathered}
+  \end{equation*}
+
+  В плохом случае, когда $2\rho''\left\Vert\mathbf{r}\right\Vert^2+\rho'<0$, можем положить $\alpha=1-\varepsilon$ --- так мы хотя бы частично учтём функцию потерь.
+\end{frame}
+
+\begin{frame}
+  \frametitle{Структура разреженности $H$}
+  Чтобы эффективно находить $\argmin\left\Vert\mathbf{J}\mathbf{x}-\mathbf{r}\right\Vert$, мы хотим понять, на каких клетках $\mathbf{J}$ и $\mathbf{H}$ могут принимать ненулевые значения.
+  \begin{itemize}
+    \item $\mathbf{J}_{i,j}\neq 0$ $\implies$ $i$-й остаток существенно зависит от $j$-го параметра
+    \item $\mathbf{H}_{j_1,j_2}\neq 0$ $\implies$ $j_1$-й и $j_2$-й параметры присутствуют вместе в некотором остатке.
+ \end{itemize}
+\end{frame}
+
+\begin{frame}
+  \frametitle{Структура разреженности $H$}
+  \begin{figure}
+    \includegraphics[height=0.7\textheight, keepaspectratio]{h-struct.png}
+    \caption*{A-E --- кадры; 1-4 --- наблюдаемые точки; K1, K2 --- параметры камер}
+  \end{figure}
+\end{frame}
+
+\begin{frame}
+  \frametitle{Структура разреженности $H$}
+  \framesubtitle{Дополнение Шура}
+  \begin{columns}
+    \begin{column}{0.5\textwidth}
+      Предположим, что мы решаем задачу SfM и пытаемся определить положения камер и ключевых точек. В простейшем случае $\mathbf{H}$ имеет вид
+      $$
+      \mathbf{H}=
+      \begin{pmatrix}
+        \begin{array}{c|c}
+          \mathbf{H}_{cc} & \mathbf{H}_{cp} \\
+          \hline
+          \mathbf{H}_{cp}^\intercal & \mathbf{H}_{pp}
+        \end{array}
+      \end{pmatrix}
+      $$
+
+    \end{column}
+    \begin{column}{0.5\textwidth}
+      \begin{figure}
+        \includegraphics[height=0.7\textheight, keepaspectratio]{h-struct-easy.png}
+        \caption*{Структура рассматриваемой $\mathbf{H}$}
+      \end{figure}
+    \end{column}
+  \end{columns}
+\end{frame}
+
+\begin{frame}
+  \frametitle{Структура разреженности $H$}
+  \framesubtitle{Дополнение Шура}
+  Мы можем воспользоваться тем, что $\mathbf{H}_{pp}$ легко обратить\footnote[frame]{На самом деле легко решать системы вида $\mathbf{H}_{pp}\mathbf{x}=\mathbf{b}$}.
+  $$
+  \begin{cases}
+    \mathbf{H}_{cc}\mathbf{x}_c+\mathbf{H}_{cp}\mathbf{x}_p=\mathbf{b}_c \\
+    \mathbf{H}_{cp}^\intercal\mathbf{x}_c+\mathbf{H}_{pp}\mathbf{x}_p=\mathbf{b}_b
+  \end{cases}
+  $$
+  \begin{equation*}
+    \begin{gathered}
+    \mathbf{x}_{p}=\mathbf{H}_{pp}^{-1}\mathbf{b}_p-\mathbf{H}_{pp}^{-1}\mathbf{H}_{cp}^\intercal\mathbf{x}_c \\
+       \left(\mathbf{H}_{cc}-\mathbf{H}_{cp}\mathbf{H}_{pp}^{-1}\mathbf{H}_{cp}^\intercal\right)\mathbf{x}_c=\mathbf{b}_c-\mathbf{H}_{cp}\mathbf{H}_{pp}^{-1}\mathbf{b}_p \\
+    \end{gathered}
+  \end{equation*}
+\end{frame}
+
+% \begin{frame}
```

Comment:
If it is not needed -- consider removing "dead" code from the file; otherwise -- consider completing this slide